src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -55,5 +55,28 @@ public void GetValueOrDefault_KeyDoesntExist_ReturnsDefaultValue()
             Assert.Equal(default(TValue), dictionary.GetValueOrDefault(key));
             Assert.Equal(defaultValue, dictionary.GetValueOrDefault(key, defaultValue));
         }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_TryAdd(int count)
+        {
+            var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+
+            TKey key = CreateTKey(seed: count);
+            TValue value = dictionary.GetValueOrDefault(key);
+            int originalCount = dictionary.Count;
+            bool alreadyContainsKey = dictionary.ContainsKey(key);
```

Comment:
You're right, it's probably not wise to have code that we're unsure will run. Updated.